src/uthash.h

```diff
@@ -264,17 +295,59 @@ do {
 } while (0)
 #endif
 
-#define HASH_ADD_KEYPTR_BYHASHVALUE_INORDER(hh,head,keyptr,keylen_in,hashval,add,cmpfcn) \
+#if HASH_NOMEM_OK
+
+#define HASH_ADD_TO_TABLE(hh,head,keyptr,keylen_in,hashval,add,op,mem_ok)        \
+do {                                                                             \
+  if ((mem_ok)) {                                                                \
+    unsigned _ha_bkt;                                                            \
+    (head)->hh.tbl->num_items++;                                                 \
+    HASH_TO_BKT(hashval, (head)->hh.tbl->num_buckets, _ha_bkt);                  \
+    HASH_ADD_TO_BKT((head)->hh.tbl->buckets[_ha_bkt], hh, &(add)->hh, mem_ok);   \
+    if (!mem_ok) {                                                               \
+      HASH_DELETE_IFBUCKET(hh,head,&(add)->hh,0);                                \
+      (add)->hh.tbl = NULL;                                                      \
+      uthash_mem_failed(add);                                                    \
+    } else {                                                                     \
+      HASH_BLOOM_ADD((head)->hh.tbl, hashval);                                   \
+      HASH_EMIT_KEY(hh, head, keyptr, keylen_in);                                \
+    }                                                                            \
+  } else {                                                                       \
+    (add)->hh.tbl = NULL;                                                        \
+    uthash_mem_failed(add);                                                      \
+  }                                                                              \
+  HASH_FSCK(hh, head, op);                                                       \
```

Comment:
Nit: This `HASH_FSCK` should probably be hoisted out into the callers of `HASH_ADD_TO_TABLE`, to avoid repeating it on lines 319 and 332. I also wonder whether avoiding the expensive `HASH_FSCK` in here would allow you to use some part of `HASH_ADD_TO_TABLE` in the implementation of `HASH_SELECT`...?